### ISSUE TITLE
polymer lint does not report when a custom element is registered by a function that is not run.

### ISSUE THREAD

**Issue by coreyfarrell** [Polymer/polymer-linter#72]
Take the following example dom-module:
```
<dom-module id="my-ele">
  <template>content</template>
  <script>
    (function() {
       'use strict';

       class MyEle extends Polymer.Element {
         static get is() { return 'my-ele'; }
       }

       window.customElements.define(MyEle.is, MyEle);
    });
  </script>
</dom-module>
```

It took me a long time to realize why one of my custom elements was not working. The anonymous function containing the element registration is not actually run.  This should produce some kind of warning as 'my-ele' is never registered.

**Comment by coreyfarrell** [comment]
It was suggested by @rictic that I use eslint to catch this error, unfortunately eslint doesn't catch this either.  Related bug report: https://github.com/eslint/eslint/issues/8385

**Comment by coreyfarrell** [comment]
I've received feedback from the eslint issue, to catch this error you need to use the [no-unused-expressions](http://eslint.org/docs/rules/no-unused-expressions) rule.

eslint is now part of my `make check` rule, so I no longer need this fixed.  Maybe just need documentation to suggest that `eslint` / other tools be used in conjunction with `polymer lint`?  Or maybe I just missed that documentation.

**Comment by rictic** [comment]
+1, added mention of eslint and htmllint to README.
